WordPress/src/main/java/org/wordpress/android/ui/prefs/AppSettingsFragment.java

```diff
@@ -145,12 +145,7 @@ private void changeLanguage(String languageCode) {
         conf.locale = newLocale;
         res.updateConfiguration(conf, res.getDisplayMetrics());
 
-        if (LanguageUtils.getCurrentDeviceLanguage(WordPress.getContext()).equals(newLocale)) {
-            // remove custom locale key when original device locale is selected
-            mSettings.edit().remove(LANGUAGE_PREF_KEY).apply();
-        } else {
-            mSettings.edit().putString(LANGUAGE_PREF_KEY, newLocale.toString()).apply();
-        }
+        mSettings.edit().putString(LANGUAGE_PREF_KEY, newLocale.toString()).apply();
```

Comment:
Do we really want to set the app locale even when it's the same as the device? That seems like it would result in unnecessary work for the app since `WPActivityUtils.applyLocale()` will always apply the locale [here](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/4932-persist-selected-locale/WordPress/src/main/java/org/wordpress/android/util/WPActivityUtils.java#L98). This could lead to more issues since having a custom locale for the app has been the source of quite a few problems.